src/Widgets/NotificationsList.vala

```diff
@@ -37,17 +37,43 @@ public class Notifications.NotificationsList : Gtk.ListBox {
         placeholder_style_context.add_class (Gtk.STYLE_CLASS_DIM_LABEL);
 
         activate_on_single_click = true;
-        margin_top = 2;
         selection_mode = Gtk.SelectionMode.NONE;
-        vexpand = true;
         set_placeholder (placeholder);
         show_all ();
 
         row_activated.connect (on_row_activated);
     }
 
     public void add_entry (NotificationEntry entry) {
-        var app_entry = add_entry_internal (entry);
+        AppEntry? app_entry = null;
+        if (entry.notification.app_info != null && entry.notification.app_info.get_id () != null) {
+            string[] desktop_id_list = {};
+            app_entries.foreach ((_app_entry) => {
+                var app_id = _app_entry.app_info.get_id ();
+
+                desktop_id_list += app_id;
+
+                if (app_id == entry.notification.desktop_id && app_entry == null) {
+                    app_entry = _app_entry;
+                }
+            });
+
+            if (!(entry.notification.desktop_id in desktop_id_list)) {
```

Comment:
Isn't this equivalent to `app_entry == null`?  We have already scanned the desktop id list.  If it had contained the notification desktop_id then app_entry would have been set.

Reply:
Yep, you're right. Thanks for helping me make sense of this :sweat_smile: